cmd/descheduler/app/descheduler_test.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/karmada-io/karmada/cmd/descheduler/app/options"
+)
+
+func TestNewDeschedulerCommand(t *testing.T) {
+	stopCh := make(chan struct{})
+	cmd := NewDeschedulerCommand(stopCh)
+
+	assert.NotNil(t, cmd)
+	assert.Equal(t, "karmada-descheduler", cmd.Use)
+	assert.NotEmpty(t, cmd.Long)
+}
+
+func TestDeschedulerCommandFlagParsing(t *testing.T) {
```

Comment:
This test case feels good 👍

Reply:
I tried to keep it maintainable and not prone to break with minor changes.